### PR TITLE
Changes in actions.yml

### PR DESCRIPTION
**Pull request by balandi1**
- Changes done to check whether workflow succeeds.

### CODE REVIEW

File: .github/workflows/actions.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ on:
 
 jobs:
   build-and-test:
-    name: Build and Test
+    name: Build and Test Server
```
Comment (pradykaushik) [inline .github/workflows/actions.yml]: **Build and Test** is okay.

File: .github/workflows/actions.yml [hunk 2]
```diff
@@ -22,5 +22,9 @@ jobs:
     - name: Get Dependencies using Go Modules
       run: go mod download
```
Comment (pradykaushik) [inline .github/workflows/actions.yml]: `make dependency`

File: .github/workflows/actions.yml [hunk 3]
```diff
@@ -22,5 +22,9 @@ jobs:
...
 
-    - name: Build Coderunner
-      run: go build -o coderunner
+    - name: Build Server
+      run: go build -o runserver
```
Comment (pradykaushik) [inline .github/workflows/actions.yml]: `make`

File: .github/workflows/actions.yml [hunk 4]
```diff
@@ -22,5 +22,9 @@ jobs:
...
+      
+    - name: Run and test server
```
Comment (pradykaushik) [inline .github/workflows/actions.yml]: **Test Server**

File: .github/workflows/actions.yml [hunk 5]
```diff
@@ -22,5 +22,9 @@ jobs:
...
+      run: ./runserver -port '8083' & go test ./...
```
Comment (pradykaushik) [inline .github/workflows/actions.yml]: For now, this is okay. Enhance the test code to accept command-line arguments using the example below.
```go
package server

import (
  "flag"
  "testing"
)

var portNumber = flag.String("port", "8082", "Port number for server to listen on")

func TestSendZip(t *testing.T) {
  port := *portNumber
  // pass port along to sendZip().
}
```

The command to test would then look like `go test ./... -args -port=8082`

**Review by pradykaushik — CHANGES_REQUESTED** [review]

File: .github/workflows/actions.yml (hunk 1 above)
Comment (balandi1) [reply .github/workflows/actions.yml]: Done

File: .github/workflows/actions.yml (hunk 2 above)
Comment (balandi1) [reply .github/workflows/actions.yml]: Done

File: .github/workflows/actions.yml (hunk 3 above)
Comment (balandi1) [reply .github/workflows/actions.yml]: Done

File: .github/workflows/actions.yml (hunk 4 above)
Comment (balandi1) [reply .github/workflows/actions.yml]: Done

File: .github/workflows/actions.yml (hunk 5 above)
Comment (balandi1) [reply .github/workflows/actions.yml]: Added to server_test.go

File: .github/workflows/actions.yml [hunk 6]
```diff
@@ -20,7 +20,11 @@ jobs:
...
-      run: go mod download
+      run: make dependency
 
-    - name: Build Coderunner
-      run: go build -o coderunner
+    - name: Build Server
+      run: make
+      
+    - name: Test server
+      run: ./code-runner-server -port '8083' & go test ./... -port '8083'
```
Comment (pradykaushik) [inline .github/workflows/actions.yml]: Use the default port as there is no valid reason for using a port other than the default one (8082).

**Review by pradykaushik — CHANGES_REQUESTED** [review]

File: .github/workflows/actions.yml (hunk 6 above)
Comment (balandi1) [reply .github/workflows/actions.yml]: Yes. That's unnecessary. Done

**Review by pradykaushik — APPROVED** [review]

**Pull request merged**